test/functional/p2p_handshake.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test P2P behaviour during the handshake phase (VERSION, VERACK messages).
+"""
+import itertools
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.messages import (
+    NODE_NETWORK,
+    NODE_NONE,
+    NODE_P2P_V2,
+    NODE_WITNESS,
+)
+from test_framework.p2p import P2PInterface
+
+
+# usual desirable service flags for outbound non-pruned peers
+DESIRABLE_SERVICE_FLAGS = NODE_NETWORK | NODE_WITNESS
```

Comment:
b6c7ae6: shouldn't we use `NODE_NETWORK_LIMITED | NODE_WITNESS` for pruned peers debug log in 49fbd57e.